charts/ceph-csi-cephfs/templates/groupsnapshotclass.yaml

```diff
@@ -0,0 +1,30 @@
+{{- if .Values.volumeGroupSnapshotClass.create -}}
+apiVersion: groupsnapshot.storage.k8s.io/v1beta1
```

Comment:
create it only if the CRD exists not only if its set to true?

Reply:
Most likely not possible and maybe out of scope, my example: https://github.com/alvistack/ansible-collection-kubernetes/blob/master/playbooks/70-helm_csi_cephfs-install.yml#L71-L90

Since we are not EMBEDDED snapshot-controller into ceph-csi, we couldn't simply check if corresponding CRD already exists by using Helm syntax?

Reply:
its possible with helm capabilities https://v2.helm.sh/docs/chart_template_guide/#built-in-objects we have done this earlier as well  For example https://github.com/ceph/ceph-csi/blob/v3.6.1/charts/ceph-csi-rbd/templates/csidriver-crd.yaml#L1

Reply:
if we do the check we dont need `.create` flag

Reply:
> its possible with helm capabilities https://v2.helm.sh/docs/chart_template_guide/#built-in-objects we have done this earlier as well For example https://github.com/ceph/ceph-csi/blob/v3.6.1/charts/ceph-csi-rbd/templates/csidriver-crd.yaml#L1

@Madhu-1 may you suggest the corresponding syntax similar as `{{ if semverCompare ">=1.18.0-beta.1" .Capabilities.KubeVersion.Version }}` for me? Since I am just new with Helm Chart templating so not much idea about that :-(

Reply:
> its possible with helm capabilities https://v2.helm.sh/docs/chart_template_guide/#built-in-objects we have done this earlier as well For example https://github.com/ceph/ceph-csi/blob/v3.6.1/charts/ceph-csi-rbd/templates/csidriver-crd.yaml#L1

I give a check with https://github.com/ceph/ceph-csi/blob/v3.14.0/charts/ceph-csi-rbd/templates/storageclass.yaml#L1 and seems semi-auto with `.create` flag looks much flexible and under control for daily usage?

Or better say: yes, we could use full-auto with `{{- if .Capabilities.APIVersions.Has "groupsnapshot.storage.k8s.io/v1beta1" }}`, but shall we also give similar check for `storageclass.yaml` and remove its `.create`, too?

Reply:
@hswong3i This is special because of the external CRD's where it need to be installed but the storageclasses exists by default., Lets do this only for volumesnapshot and volumegroupsnapshot only

Reply:
Thats something possible because not everyone wants all the features but we can take it in future PR as well, This is an extra check to ensure the installation doesnt fail by default if the CRD's are not present

Reply:
> Thats something possible because not everyone wants all the features but we can take it in future PR as well, This is an extra check to ensure the installation doesnt fail by default if the CRD's are not present

Totally agree and that's also how `.create` working: first KISS with true/false options as semi-auto, future PR could upgrade and improve it with full-auto automatic detect and checking.